snap/snapenv/snapenv.go

```diff
@@ -97,7 +97,7 @@ func userEnv(info *snap.Info, home string) map[string]string {
 	result := map[string]string{
 		"SNAP_USER_COMMON": info.UserCommonDataDir(home),
 		"SNAP_USER_DATA":   info.UserDataDir(home),
-		"XDG_RUNTIME_DIR":  info.UserXdgRuntimeDir(os.Geteuid()),
+		"XDG_RUNTIME_DIR":  info.UserXdgRuntimeDirGlobal(os.Geteuid()),
```

Comment:
I know we talked about this a few times, but what is the consequence of using this as the XDG_RUNTIME_DIR?  Specifically, will snaps break because they will attempt to write into this directory but they don't have permission as apparmor is blocking most paths?

Also, what about upgrades?  Will snaps expect data to be inside that directory and then magically break after an automatic refresh?